pallets/lbp/src/benchmarking.rs

```diff
@@ -0,0 +1,323 @@
+			id: ASSET_ID_A,
+			amount: BalanceOf::<T>::from(1_000_000_000_u32),
+			initial_weight: 20,
+			final_weight: 90
+		};
+		let asset_b = LBPAssetInfo{
+			id: ASSET_ID_B,
+			amount: BalanceOf::<T>::from(2_000_000_000_u32),
+			initial_weight: 80,
+			final_weight: 10
+		};
+
+		let pool_id = T::AssetPairPoolId::from_assets(asset_a.id, asset_b.id);
+
+		LBP::<T>::create_pool(RawOrigin::Root.into(), caller.clone(), asset_a, asset_b, duration, WeightCurveType::Linear, true)?;
+		ensure!(PoolData::<T>::contains_key(&pool_id), "Pool does not exist.");
+
+	}: _(RawOrigin::Signed(caller), pool_id.clone(), (asset_a.id, 500_000_000_u128), (asset_b.id, 1_000_000_000_u128))
+	verify {
+		assert_eq!(T::MultiCurrency::free_balance(asset_a.id, &pool_id), 500_000_000_u128);
+		assert_eq!(T::MultiCurrency::free_balance(asset_b.id, &pool_id), 1_000_000_000_u128);
+	}
+
+	destroy_pool {
+		let caller = funded_account::<T>("caller", 0);
+		let duration = (T::BlockNumber::from(10_u32), T::BlockNumber::from(20_u32));
+		let asset_a = LBPAssetInfo{
+			id: ASSET_ID_A,
+			amount: BalanceOf::<T>::from(1_000_000_000_u32),
+			initial_weight: 20,
+			final_weight: 90
+		};
+		let asset_b = LBPAssetInfo{
+			id: ASSET_ID_B,
+			amount: BalanceOf::<T>::from(2_000_000_000_u32),
+			initial_weight: 80,
+			final_weight: 10
+		};
+
+		let pool_id = T::AssetPairPoolId::from_assets(asset_a.id, asset_b.id);
+
+		LBP::<T>::create_pool(RawOrigin::Root.into(), caller.clone(), asset_a, asset_b, duration, WeightCurveType::Linear, true)?;
+		ensure!(PoolData::<T>::contains_key(&pool_id), "Pool does not exist.");
+
+		System::<T>::set_block_number(21u32.into());
+
+	}: _(RawOrigin::Signed(caller), pool_id.clone())
+	verify {
+		assert_eq!(PoolData::<T>::contains_key(&pool_id), false);
+	}
+
+	sell {
+		let caller = funded_account::<T>("caller", 0);
+		let duration = (T::BlockNumber::from(10_u32), T::BlockNumber::from(20_u32));
+		let asset_a = LBPAssetInfo{
+			id: ASSET_ID_A,
+			amount: BalanceOf::<T>::from(1_000_000_000_u32),
+			initial_weight: 20,
+			final_weight: 90
+		};
+		let asset_b = LBPAssetInfo{
+			id: ASSET_ID_B,
+			amount: BalanceOf::<T>::from(2_000_000_000_u32),
+			initial_weight: 80,
+			final_weight: 10
+		};
+
+		let asset_in: AssetId = ASSET_ID_A;
+		let asset_out: AssetId = ASSET_ID_B;
+		let amount : Balance = 100_000_000;
+		let max_limit: Balance = 10_000_000;
+
+		let pool_id = T::AssetPairPoolId::from_assets(asset_a.id, asset_b.id);
+
+		LBP::<T>::create_pool(RawOrigin::Root.into(), caller.clone(), asset_a, asset_b, duration, WeightCurveType::Linear, true)?;
+		ensure!(PoolData::<T>::contains_key(&pool_id), "Pool does not exist.");
+
+		LBP::<T>::unpause_pool(RawOrigin::Signed(caller.clone()).into(), pool_id.clone())?;
+		let pool_data = LBP::<T>::pool_data(&pool_id);
+		assert_eq!(pool_data.paused, false);
+
+		System::<T>::set_block_number(12u32.into());
+
+	}: _(RawOrigin::Signed(caller.clone()), asset_in, asset_out, amount, max_limit)
```

Comment:
does this benchmark take into account the weight update when validating ?  Because that would the worst case here.

Reply:
yes, weights are updated because there is only one trade event in the block (weights are updated prior the first trade, in the same extrinsic), and the block number is different than the `start` block. But I'm going to double-check it, somehow. Just to be sure.